src/tests/conformance/get_data.rs

```diff
@@ -1,129 +1,235 @@
+    // Spin up a node instance with knowledge of the initial testnet-chain.
     let mut node = Node::new().unwrap();
-    node.initial_action(Action::SeedWithTestnetBlocks(11))
+    node.initial_action(Action::SeedWithTestnetBlocks(SEED_BLOCKS.len()))
         .start()
-        .await
-        .unwrap();
-
-    // block headers and hashes
-    let blocks = vec![
-        Box::new(Block::testnet_genesis()),
-        Box::new(Block::testnet_1()),
-        Box::new(Block::testnet_2()),
-    ];
-
-    let inv_blocks = blocks
-        .iter()
-        .map(|block| block.inv_hash())
-        .collect::<Vec<_>>();
-
-    // Establish a peer node
+        .await?;
+
+    // Create a synthetic node.
     let mut synthetic_node = SyntheticNode::builder()
         .with_full_handshake()
         .with_all_auto_reply()
         .build()
-        .await
-        .unwrap();
-
-    synthetic_node.connect(node.addr()).await.unwrap();
-
-    // Query for the first i blocks
-    for i in 0..blocks.len() {
-        synthetic_node
-            .send_direct_message(
-                node.addr(),
-                Message::GetData(Inv::new(inv_blocks[..=i].to_vec())),
-            )
-            .await
-            .unwrap();
-
-        // Expect the i blocks
-        for (j, expected_block) in blocks.iter().enumerate().take(i + 1) {
-            let (_, block) = synthetic_node.recv_message_timeout(TIMEOUT).await.unwrap();
-            let block = assert_matches!(block, Message::Block(block) => block);
-            assert_eq!(block, *expected_block, "run {}, {}", i, j);
-        }
-    }
+        .await?;
 
-    // Query for a non-existant block
-    let non_existant = InvHash::new(ObjectKind::Block, Hash::new([17; 32]));
-    let non_existant_inv = Inv::new(vec![non_existant]);
+    // Connect to the node and initiate handshake.
+    synthetic_node.connect(node.addr()).await?;
 
+    // Send the query.
     synthetic_node
-        .send_direct_message(node.addr(), Message::GetData(non_existant_inv.clone()))
-        .await
-        .unwrap();
-
-    let (_, not_found) = synthetic_node.recv_message_timeout(TIMEOUT).await.unwrap();
-    let not_found = assert_matches!(not_found, Message::NotFound(not_found) => not_found);
-    assert_eq!(not_found, non_existant_inv);
-
-    // Query a mixture of existing and non-existing blocks
-    let mut mixed_blocks = inv_blocks;
-    mixed_blocks.insert(1, non_existant);
-    mixed_blocks.push(non_existant);
-
-    let expected = vec![
-        Message::Block(Box::new(Block::testnet_genesis())),
-        Message::NotFound(non_existant_inv.clone()),
-        Message::Block(Box::new(Block::testnet_1())),
-        Message::Block(Box::new(Block::testnet_2())),
-        Message::NotFound(non_existant_inv),
-    ];
+        .send_direct_message(node.addr(), query)
+        .await?;
 
+    // Send a Ping - once we receive the matching Pong we know our query has been fully processed.
+    let nonce = Nonce::default();
```

Comment:
Did we get rid of the ping/pong assertion helper on `SyntheticNode`?

Reply:
No its still there - but it only receives a single message before returning.

`GetData` elicits a `Block` message for every known hash in the request. So in this case the looping receive is needed.

One could change / extend a multi version on the `SyntheticNode`, but so far this is the only use case for it. So I opted to just leave it here.